pandas/tests/indexes/test_interval.py

```diff
@@ -657,206 +635,39 @@ def testcontains(self):
-        assert Interval(1, 3, closed='both').overlaps(Interval(-1, 1, closed='both'))
-
-        assert not Interval(1, 3, closed='both').overlaps(Interval(3, 5, closed='right'))
-        assert Interval(1, 3, closed='both').overlaps(Interval(3, 5, closed='left'))
-        assert Interval(1, 3, closed='both').overlaps(Interval(3, 5, closed='both'))
-
-    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
-    def test_interval_overlaps_intervalIndex_updated_behavior(self):
-
-        # class Interval:
-        #     def overlaps(self, other: IntervalIndex) -> IntegerArray1D
-
-        idx = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='right')
-
-        tm.assert_numpy_array_equal(Interval(1, 3, closed='right').overlaps(idx), np.array([1, 2]))
-        tm.assert_numpy_array_equal(Interval(1, 2, closed='right').overlaps(idx), np.array([2]))
-        tm.assert_numpy_array_equal(Interval(0, 2, closed='right').overlaps(idx), np.array([0, 2]))
-        tm.assert_numpy_array_equal(Interval(3, 4, closed='right').overlaps(idx), np.array([]))
-
-        tm.assert_numpy_array_equal(Interval(1, 3, closed='left').overlaps(idx), np.array([0, 1, 2]))
-        tm.assert_numpy_array_equal(Interval(1, 2, closed='left').overlaps(idx), np.array([0, 2]))
-        tm.assert_numpy_array_equal(Interval(0, 2, closed='left').overlaps(idx), np.array([0, 2]))
-        tm.assert_numpy_array_equal(Interval(3, 4, closed='left').overlaps(idx), np.array([3]))
-
-        tm.assert_numpy_array_equal(Interval(1, 3, closed='both').overlaps(idx), np.array([0, 1, 2]))
-        tm.assert_numpy_array_equal(Interval(1, 2, closed='both').overlaps(idx), np.array([0, 2]))
-        tm.assert_numpy_array_equal(Interval(0, 2, closed='both').overlaps(idx), np.array([0, 2]))
-        tm.assert_numpy_array_equal(Interval(3, 4, closed='both').overlaps(idx), np.array([3]))
-
     @pytest.mark.xfail(reason="new indexing tests for issue 16316")
-    def test_intervalIndex_covers_interval_updated_behavior(self):
+    def test_contains_updated_behavior(self):
 
-        # class IntervalIndex:
-        #     def covers(self, other: Interval) -> IntegerArray1D
+        # this method will replace both of the above
 
-        idx = IntervalIndex.from_tuples([(0, 1), (2, 3), (1, 3)], closed='right')
+        index = IntervalIndex.from_arrays([0, 1], [1, 2], closed='right')
 
-        tm.assert_numpy_array_equal(idx.covers(Interval(1, 3, closed='right')), np.array([1, 2]))
-        tm.assert_numpy_array_equal(idx.covers(Interval(0, 3, closed='right')), np.array([0, 1, 2]))
-        tm.assert_numpy_array_equal(idx.covers(Interval(0, 2, closed='right')), np.array([0]))
-        tm.assert_numpy_array_equal(idx.covers(Interval(2, 4, closed='right')), np.array([1]))
+        assert 0 not in index
+        assert 1 in index # not sure how we want to do this.
+        assert 2 in index # does this defer to "index.contains()" ?
```

Comment:
I don't think we want to do this (I think we discussed it before).

IMO we should keep `in` (`__contains__`) for checking if the exact element is in the index. 
We  can have a separate `contains` method to do such checks.